server/repository/assignments.go

```diff
@@ -0,0 +1,151 @@
+
+	duration := 0
+
+	for rows.Next() {
+		var pairID int
+		rows.Scan(&pairID)
+
+		_, err := insert.Exec(userID, pairID, experimentID, nil, duration)
+		if err != nil {
+			return nil, fmt.Errorf("DB error: %v", err)
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, fmt.Errorf("DB error: %v", err)
+	}
+
+	return repo.GetAll(userID, experimentID)
+}
+
+// Create stores an Assignment into the DB, and returns that new Assignment
+func (repo *Assignments) Create(as *model.Assignment) error {
+
+	_, err := repo.db.Exec(
+		`INSERT INTO assignments (user_id, pair_id, experiment_id, answer, duration)
+		VALUES ($1, $2, $3, $4, $5)`,
+		as.UserID, as.PairID, as.ExperimentID, as.Answer, as.Duration)
+
+	if err != nil {
+		return err
+	}
+
+	as, err = repo.Get(as.UserID, as.PairID)
+	return err
+}
+
+// getWithQuery builds a Assignment from the given sql QueryRow. If the
+// Assignment does not exist, it returns nil, nil
+func (repo *Assignments) getWithQuery(queryRow *sql.Row) (*model.Assignment, error) {
+	var as model.Assignment
+
+	err := queryRow.Scan(
+		&as.UserID, &as.PairID, &as.ExperimentID, &as.Answer, &as.Duration)
+
+	switch {
+	case err == sql.ErrNoRows:
+		return nil, nil
+	case err != nil:
+		return nil, fmt.Errorf("Error getting assignment from the DB: %v", err)
+	default:
+		return &as, nil
+	}
+}
+
+// Get returns the Assignment for the given user and pair IDs. If the Assignment
+// does not exist, it returns nil, nil
+func (repo *Assignments) Get(userID, pairID int) (*model.Assignment, error) {
+	return repo.getWithQuery(repo.db.QueryRow(
+		"SELECT * FROM assignments WHERE user_id=$1 AND pair_id=$2",
+		userID, pairID))
+}
+
+// GetAll returns all the Assignments for the given user and experiment IDs.
+// Returns an ErrNoAssignmentsInitialized if they do not exist yet
+func (repo *Assignments) GetAll(userID, experimentID int) ([]*model.Assignment, error) {
+	rows, err := repo.db.Query(
+		"SELECT * FROM assignments WHERE user_id=$1 AND experiment_id=$2",
+		userID, experimentID)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting assignments from the DB: %v", err)
+	}
+	defer rows.Close()
+
+	results := make([]*model.Assignment, 0)
+
+	for rows.Next() {
+		var as model.Assignment
+		rows.Scan(
+			&as.UserID, &as.PairID, &as.ExperimentID, &as.Answer, &as.Duration)
+
+		results = append(results, &as)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("DB error: %v", err)
+	}
+
+	if len(results) == 0 {
+		return nil, ErrNoAssignmentsInitialized
+	}
+
+	return results, nil
+}
+
+// Update updates the Assignment identified by the given user and pair IDs,
+// with the given answer and duration
+func (repo *Assignments) Update(userID int, pairID int, answer string, duration int) error {
+	if _, ok := model.Answers[answer]; !ok {
+		return fmt.Errorf("Wrong answer provided: '%s'", answer)
+	}
+
+	cmd := fmt.Sprintf(
+		"UPDATE assignments SET answer='%v', duration=%v WHERE user_id=%v AND pair_id=%v",
```

Comment:
>_We considered that the same `FilePair` can be assigned to the same `User` many times in different experiments, so it can not be `PRIMARY KEY`_

The idea of checking the `user_id` is redundant if it is used the `assignmentId`, but it can be a good idea to ensure that the assignment being saved, belongs to the user in session.